File: angular/src/app/models/recipe.ts
import { Category } from "./category";
import { Difficulty } from "./difficulty";
import { Feedback } from "./feedback";
import { Ingredient } from "./ingredient";
import { Instruction } from "./instruction";
import { Like } from "./like";
import { User } from "./user";


export class Recipe {
  id!: number;
  title!: string | null;
  description!: string | null;
  videoUrl!: string | null;
  servings!: number | null;
  prepTime!: number | null;
  calories!: number | null;
  fat!: number | null;
  protein!: number | null;
  carbs!: number | null;
  userId!: number | null;
  likes!: Like[] | null;
  difficultyId!: number | null;
  status!: number | null;
  createAt!: Date | null;
  updateAt!: Date | null;
  categoryId!: number | null;
  category!: Category | null;
  difficulty!: Difficulty | null;
  feedbacks!: Feedback[];
  imageUrl!: string | null;
  ingredients!: Ingredient[];
  instructions!: Instruction[];
  user!: User | null;
  countLikes: number = 0;
}

File: DAL8/Models/DB/Recipe.cs
using System;
using System.Collections.Generic;

namespace DAL.Models.DB;

public partial class Recipe
{
    public int Id { get; set; }

    public string? Title { get; set; }

    public string? Description { get; set; }

    public string? VideoUrl { get; set; }

    public double? Servings { get; set; }

    public double? PrepTime { get; set; }

    public double? Calories { get; set; }

    public double? Fat { get; set; }

    public double? Protein { get; set; }

    public double? Carbs { get; set; }

    public int? UserId { get; set; }

    public int? DifficultyId { get; set; }

    public int? Status { get; set; }

    public DateTime? CreateAt { get; set; }

    public DateTime? UpdateAt { get; set; }

    public int? CategoryId { get; set; }
    public string? ImageUrl { get; set; }

    public virtual Category? Category { get; set; }

    public virtual Difficulty? Difficulty { get; set; }
    public virtual ICollection<Likes> Likes { get; set; } = new List<Likes>();

    public virtual ICollection<Feedback> Feedbacks { get; set; } = new List<Feedback>();

    public virtual ICollection<Ingredient> Ingredients { get; set; } = new List<Ingredient>();
    public virtual ICollection<Instruction> Instructions { get; set; } = new List<Instruction>();

    public virtual User? User { get; set; }
}
